lambdas/data-migration1/tests/test-rules.js

```diff
@@ -0,0 +1,394 @@
+        updated_at: new Date(fakeRule.updatedAt),
+      },
+      ['createdAt', 'updatedAt', 'state', 'provider', 'collection', 'rule']
+    )
+  );
+});
+
+test.serial('migrateRuleRecord ignores extraneous fields from Dynamo', async (t) => {
+  const { knex, providerKmsKeyId } = t.context;
+  fakeCollection = fakeCollectionFactory();
+  fakeProvider = fakeProviderFactory({
+    encrypted: false,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  });
+  const fakeRule = generateFakeRule(fakeCollection.name, fakeCollection.version, fakeProvider.id);
+
+  fakeRule.state = 'ENABLED';
+  await migrateCollectionRecord(fakeCollection, knex);
+  await migrateProviderRecord(fakeProvider, providerKmsKeyId, knex);
+  await t.notThrowsAsync(migrateRuleRecord(fakeRule, knex));
+});
+
+test.serial('migrateRuleRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+  const { knex, providerKmsKeyId } = t.context;
+  const fakeRule = generateFakeRule(fakeCollection.name, fakeCollection.version, fakeProvider.id);
+
+  await migrateCollectionRecord(fakeCollection, knex);
+  await migrateProviderRecord(fakeProvider, providerKmsKeyId, knex);
+  await migrateRuleRecord(fakeRule, knex);
+  await t.throwsAsync(
+    migrateRuleRecord(fakeRule, t.context.knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migrateRules skips already migrated record', async (t) => {
+  const { knex, providerKmsKeyId } = t.context;
+  const fakeRule = generateFakeRule(fakeCollection.name, fakeCollection.version, fakeProvider.id);
+  const queueUrls = await createSqsQueues(randomString());
+  fakeRule.queueUrl = queueUrls.queueUrl;
+
+  await migrateCollectionRecord(fakeCollection, knex);
+  await migrateProviderRecord(fakeProvider, providerKmsKeyId, knex);
+  await migrateRuleRecord(fakeRule, knex);
+  await rulesModel.create(fakeRule);
+
+  t.teardown(() => rulesModel.delete(fakeRule));
+  const migrationSummary = await migrateRules(process.env, knex);
+  t.deepEqual(migrationSummary, {
+    dynamoRecords: 1,
+
+    skipped: 1,
+    failed: 0,
+    success: 0,
+  });
+  const records = await t.context.knex.queryBuilder().select().table('rules');
+  t.is(records.length, 1);
+});
+
+test.serial('migrateRules processes multiple rules', async (t) => {
+  const { knex, providerKmsKeyId } = t.context;
+  const anotherFakeCollection = fakeCollectionFactory();
+  const anotherFakeProvider = fakeProviderFactory({
+    encrypted: false,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  });
+  const fakeRule1 = generateFakeRule(fakeCollection.name, fakeCollection.version, fakeProvider.id);
+  const fakeRule2 = generateFakeRule(anotherFakeCollection.name, anotherFakeCollection.version, anotherFakeProvider.id);
+  const queueUrls1 = await createSqsQueues(randomString());
```

Comment:
I don't think we need to use this helper, which actually creates an SQS queue in Localstack. We can just use random strings for the queue URLs, since that's all the field validation requires.